test/python/result/generate_data.py

```diff
@@ -0,0 +1,294 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=invalid-name
+
+"""Data generator for readout mitigation tests"""
+from numpy import array
+from qiskit import QiskitError
+
+try:
+    from qiskit import Aer
```

Comment:
This isn't correct all this does is check that the lazy wrapper is present and will always evaluate true. Instead check that you can import aer with:

```suggestion
    from qiskit.providers.aer import Aer
```